### ISSUE TITLE
Some GameParameters.cfg values not applied

### ISSUE THREAD

**Issue by lpgagnon** [KSP-RO/RP-0#1126]
First noticed with the Range Modifier: RP-1 Moderate Difficulty tries to set it to 1 but starting a new career and choosing Moderate sets it to 0.8 instead (weirdly, both Normal and Hard get 1.0)
https://github.com/KSP-RO/RP-0/blob/master/GameData/RP-0/GameParameters.cfg#L158

buildingimpactmultiplier and occlusionModifier values likewise don't match between .cfg and actual value.

Might actually be _all_ values that are affected; I haven't tried seeing if values that currently match still do so if the .cfg values are edited. But presumably the startingfunds is working properly...

**Comment by siimav** [comment]
Some values cannot be set through that GameParameters file and anything related to CommNet is apparently also affected. Actually RSS is already setting some of those values by itself: https://github.com/KSP-RO/RealSolarSystem/blob/master/Source/CommNetFixer.cs#L38-L39

**Comment by Gordon-Dry** [comment]
I already mentioned the fact about occlusionModifier in the RO / RP-1 Discord channel some days ago.
So I'm not alone...

**Comment by Gordon-Dry** [comment]
@siimav and RSS does that by code? Damn it.

**Comment by MikeOnTea** [comment]
This is a long known problem, and i think i already dug throug the code a bit a while ago with no results.
If KSP doesn't allow to set these settings, we might have to go the RSS way and just override them.

**Comment by Gordon-Dry** [comment]
But with using an UI should be possible.
As game core allows to move sliders and then apply the new values, it should be possible for mods, too.
And all this also should allow using values in a config, but it should be read by the mod code and applied then instead of trying to MM patch them.
The config could resist inside a PluginData folder to avoid MM trying to parse it - or it's no .cfg anymore.

**Comment by siimav** [comment]
Currently the implementation in RSS will always overwrite any change the user may have made to the settings. It's probably possible to create a custom ScenarioModule that would run only when the space center scene is loaded for the first time and then override the commnet values with whatever we want. However I have no idea whether RSS is the correct place for that or should it be in RO instead?

**Comment by DRVeyl** [comment]
Annoying that RSS does this in code, but a much simpler solution is to politely ask it not to in the configs.  RSS/RealSolarSystem.cfg builds the root node:

REALSOLARSYSTEM
{
    overrideCommNetParams = True
    ... <stuff>
}

Just patch that to False, and the RSS CommNetFixer module referenced above will do nothing.

**Comment by lpgagnon** [comment]
I've tried overrideCommNetParams=false, and that's a false lead. The overridden values I originally saw (rangeModifier=0.8, etc) don't come from CommnetFixer, but from something else (possibly just the base game ignoring the GameParameters values for those fields). If the values this produces aren't good for RP-0, something like CommnetFixer would seem necessary (unless GameParameters is just using the wrong field names).

CommnetFixer does allow specifying occlusion values via @REALSOLARSYSTEM if for some reason its default 1.0 isn't good enough; however it doesn't deal with the Range Modifier value (which was the original issue I was reporting). Either RP-0 needs its own CommnetFixer, or (more likely) RSS's CommnetFixer should be extended to handle rangeModifier as well. (except there's non-commnet values RP-0 tries to set that also get ignored)

2 obvious limitations of the CommnetFixer approach:
- no support for per-difficulty-setting values. (maybe it could be extended to load the GameParamers object and the current difficulty?)
- if the user _manually_ changes the values, CommnetFixer still overrides them. (this seems to happen once per game session, when loading/creating a career - confusing at best)

**Comment by pap1723** [comment]
Is this still something that needs to be looked at with RealAntennas? @DRVeyl @lpgagnon

**Comment by lpgagnon** [comment]
Nothing has changed, so yes. Note that comms aren't the only thing affected; I mentioned buildingimpactmultiplier

**Comment by DRVeyl** [comment]
The issue is that *Squad* sets them in code.  They are overriding any configured settings when they implement `public override void SetDifficultyPreset(GameParameters.Preset preset)` for the respective parameter class.

The CommNetParams class has configured difficulty presets and has hardcoded the values as:
```
        case GameParameters.Preset.Moderate:
          rangeModifier = 0.8f;
          occlusionMultiplierVac = 1f;
          occlusionMultiplierAtm = 0.85f;
          plasmaBlackout = false;
          break;
``` 
So for these parameters, at least, KSP is uninterested in those repsective values in the `GAMEPARAMETERS` ConfigNode.

Similar for a few Advanced Params:
```
        case GameParameters.Preset.Moderate:
          BuildingImpactDamageMult = 0.1f;
          AllowNegativeCurrency = true;
          ResourceTransferObeyCrossfeed = true;
```

Why they would do this, when everything else appears to happily support the ConfigNode, I have no idea.  And I suppose this hasn't made it from mods' GitHubs to the Squad BugTracker.

**Comment by DRVeyl** [comment]
Well, now it has, with <https://bugs.kerbalspaceprogram.com/issues/25962>.  Maybe for 1.11!

**Comment by NathanKell** [comment]
@DRVeyl does this interfere with RA at all? Is it worth me fixing with Harmony in RP-1?

**Comment by DRVeyl** [comment]
No.  Had to go look at the code, but starting with RA 2.0 it no longer cares about those difficulty settings.  The radius for a celestial body occluder is the body's radius.  <https://github.com/DRVeyl/RealAntennas/blob/master/src/RealAntennasProject/Precompute/Precompute.cs#L701>
